### PR TITLE
Implement design specification

### PR DESCRIPTION
**Pull request by ccharp**

### CODE REVIEW

**Review by nickjsr — COMMENTED** [review]
Good work.
I believe we should include detailed information regarding each pipeline method and the interactions between them.

**Comment by nickjsr** [comment]
I pushed a new commit to main directly because all my changes have been reviewed by both of you. The changes are adding slice_test.py, removing pipeline_test, and fixing a small pylint issue. Thank you!

File: docs/design_spec.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+##Software Design
+The overall design is quite simple: we have a single class representing a dataframe and a module containing all pipeline methods. 
+
+###Components
+
+####Package: `dplypy`
+Since the two modules, dplyframe and pipeline rely on each other to provide the combined functionality of Dplyr, the package is imported with a single statement, that is, they cannot be imported separately. 
+
+#####Module: `dplyframe.py`
+DplyFrame represents the data we want to transform. The DplyFrame is water that flows through the pipeline. Behaviorally, it almost entirely mirrors `Pandas.DataFrame`, excepting the (`+`) operatore which allows pipeline methods to be chained together.
```
Comment (MatthewChau99) [inline docs/design_spec.md:10]: "operatore" -> "operator"

**Review by MatthewChau99 — APPROVED** [review]

**Comment by nickjsr** [comment]
I highly appreciate your update Corbin! Again, I believe we should enumerate all pipeline methods with some tiny details. Thanks!

**Review by nickjsr — APPROVED** [review]

**Pull request merged**